### PR TITLE
Add feature `drag and drop` to upload files

### PR DESCRIPTION
**Pull request by kuzmavictor**
During this iteration was implemented:

- feature `drag and drop` to upload files(the class that implements it is located on the following path: `name_application\view\componts\draggable-component` );

- the necessary functionality inside  (`FileList`, `File-Item`);

- the styles for checking work.

### CODE REVIEW

**Comment by kuzmavictor** [comment]
@dkalinin PTAL.

File: app/service/store/action/index.js [hunk 1]
```diff
@@ -33,4 +33,20 @@ export const actions = {
...
     });
   },
+  /**
+   * Adds files to file list. The promise that returns emulates working with server-side.
+   *
+   * @param {Store} store - an instance of {@link Store}
+   * @return {function(void)}
+   */
+  addFiles: (store) => (fileList, folderId) => {
+    store.commit('setFilesLoadingStatus', false);
```
Comment (dkalinin) [inline app/service/store/action/index.js]: isFileListLoading

File: app/service/store/mutation/index.js [hunk 2]
```diff
@@ -33,4 +33,30 @@ export const mutations = {
...
+   * Adds files to file list.
+   *
+   * @param {String} state - a state name
+   * @param {File} files - an uploading file
+   * @param {String} folderId - an identifier of folder
+   *
+   * @see {@link FileList}
+   */
+  addFiles(state, files, folderId) {
+    const currantFileList = state.files;
```
Comment (dkalinin) [inline app/service/store/mutation/index.js]: currant -> current

File: app/service/store/pub-sub/index.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ export class PubSub {
   }
 
   /**
-   * Provides functionality of publishing event(returns the callbacks for an event)
+   * Provides functionality of publishing event(returns the callbacks for an event).
```
Comment (dkalinin) [inline app/service/store/pub-sub/index.js]: Publishes event.

File: app/view/component/draggable-component/index.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  /**
+   * Occurs when the user releases a roaming item over a folder
+   *
+   * @param {DragEvent} event - a DOM event that represents a drag and drop interaction
+   * @param {function}callback - a callback function.
+   * @param {String} id - an id of file
+   */
+  onDropToFolder(event, callback, id) {
```
Comment (dkalinin) [inline app/view/component/draggable-component/index.js]: This should be in FileItem.

File: app/view/component/draggable-component/index.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    callback(Array.from(event.dataTransfer.files), id);
+    this.onDragLeave(event, this._htmlDivElement);
+  }
+
+  /**
+   * Occurs when the user releases a roaming item over a file.
+   *
+   * @param {DragEvent} event - a DOM event that represents a drag and drop interaction
+   */
+  onDropToList(event) {
```
Comment (dkalinin) [inline app/view/component/draggable-component/index.js]: This should be in FileList.

File: app/view/component/file-list/index.js [hunk 6]
```diff
@@ -25,27 +25,50 @@ export class FileList extends ConfigComponent {
...
+      this.state.files.forEach((file) => {
+        const fileItem = new FileItem(this._htmlUlElement, file.name, file.id, file.type);
+
         fileItem.onClickDelete(() => {
           fileItem.disabledButtonElement = true;
-          this.dispatch('removeItem', element.id);
+          this.dispatch('removeItem', file.id);
+        });
+
+        fileItem.onFolderDrop = ((files, folderId)=> {
```
Comment (dkalinin) [inline app/view/component/file-list/index.js]: `fileItem.onFolderDrop = (` -> `fileItem.onFolderDrop(`

File: app/view/component/file-list/index.js [hunk 7]
```diff
@@ -25,27 +25,50 @@ export class FileList extends ConfigComponent {
...
     }
   }
 
   /**
-   * Creates span element that provide rendering process of loading file-list
+   * Creates span element that provide rendering process of loading file list.
+   */
+  _createLoadingSpan() {
+    this._loadingSpan = new Span(this._root);
+    this._loadingSpan.id = 'loading-span';
```
Comment (dkalinin) [inline app/view/component/file-list/index.js]: `id` is not needed.

**Comment by kuzmavictor** [comment]
@dkalinin fixed, PTAL.

File: app/service/store/pub-sub/index.js [hunk 8]
```diff
@@ -12,7 +12,7 @@ export class PubSub {
   }
 
   /**
-   * Provides functionality of subscribing new event with callback (adding new event with callback).
+   * Subscribes on en event (adding new event with callback).
```
Comment (dkalinin) [inline app/service/store/pub-sub/index.js]: en -> an.

**Pull request merged**